CRM/Contribute/Form/UpdateBilling.php

```diff
@@ -203,27 +203,28 @@ public function postProcess() {
     $processorParams['month'] = CRM_Core_Payment_Form::getCreditCardExpirationMonth($processorParams);
     $processorParams['year'] = CRM_Core_Payment_Form::getCreditCardExpirationYear($processorParams);
     $processorParams['subscriptionId'] = $this->getSubscriptionDetails()->processor_id;
-    $processorParams['amount'] = $this->_subscriptionDetails->amount;
+    $processorParams['amount'] = $this->getSubscriptionDetails()->amount;
+    $processorParams['id'] = $this->getContributionRecurID();
```

Comment:
UpdateSubscription didn't used to pass the ID of the contribution recur but was fixed some time ago. UpdateBilling still hasn't been fixed and it's crazy not to pass the ID when we have it!

Reply:
@mattwire if we are adding this we should use the parameter name in the documentation - ie 
` $processorParams['contributionRecurID'] `

I'm OK with you passing additional names as well - but we should make sure we pass out the parameters we have contracted to in all cases

 https://docs.civicrm.org/dev/en/latest/extensions/payment-processors/create/#recurring-contribution-parameters

Reply:
@eileenmcnaughton Good idea. I've updated both UpdateBilling and UpdateSubscription to pass both `id` and `contributionRecurID` so we are consistent.

Reply:
@mattwire I see this class as more correct because it builds up a separate 'processorParams' - the other class might be storing id in params for 'other reasons'?